spacewalk/setup/share/embedded_diskspace_check.py

```diff
@@ -128,8 +125,8 @@ def getNeeds(needsDict=None):
     """
 
     needsDict = needsDict or DEFAULT_NEEDS
-    mp2pMap = paths2mountpoints(needsDict.keys())[1]
-    mp2fsMap = paths2freespace(mp2pMap.keys())
+    mp2pMap = paths2mountpoints(list(needsDict.keys()))[1]
+    mp2fsMap = paths2freespace(list(mp2pMap.keys()))
```

Comment:
I want to ensure we do preserve `list` as the expected output from `dict.keys()`. Switching to `tuple` might introduce problems in case those invoked functions are expecting a list. I tend to be conservative here.